src/fileimpl.cpp

```diff
@@ -369,15 +369,13 @@ namespace zim
 
   std::shared_ptr<const Dirent> FileImpl::getDirentByTitle(article_index_t idx)
   {
-    if (idx >= getCountArticles())
-      throw ZimFileFormatError("article index out of range");
     return getDirent(getIndexByTitle(idx));
   }
 
   article_index_t FileImpl::getIndexByTitle(article_index_t idx)
```

Comment:
I have added the `title_index_t`.
As said in the commit, I've not added `title_index_type` as it is not a real typedef and it
would not help us here.

In fact, I even ask myself if we can simply remove the public methods `getEntryBy*(entry_index_type idx)`. I don't see a use case where we would like to get the `nth` article in a zim. It could be useful to iterate over the articles, but we have iterator for that.